interview-questions.md

```diff
@@ -8,44 +8,46 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is a framework built to run on the programming language Ruby, which allows us to create and work with databases that can be connected to our front-end applications. Although I'm relatively new to using Rails, I know that there's a LOT of code at it's core, and the real 'magic' that makes it so popular is how it manages to effortlessly perform CRUD actions with very little code, so long as we follow a few syntactical rules.
 
-Researched answer:
+Researched answer: In addition to my previous response, Rails uses the MVC (Model View Controller) pattern as its' architecture. When it generates an application, the Model is the data structure, the View is the displayed output on a webpage, and the Controller is what connects the two.
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is a database where the values have some kind of connection to each other. Relational databases take on the form of what we're traditionally familiar with as tables, where each column provides a descriptive variable that we can then fill up with a row of data that pertains to each column. So, in a sense, the data is all related, and we could even have multiple tables with related data. There ARE non-relational databases, which don't take on the traditional look of a table and will typically just have a piece of data stored with some kind of primary key to access it with. Non-relational databases are great for storing things in more of a key:value format.
 
-Researched answer:
+Researched answer: In addition to my previous answer, relational databases are also referred to as SQL databases. SQL is short for Structured Query Language, and it's what most relational databases are written in.
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are unique identifiers that are assigned to a column within your database, and used to make sure that every instance of a row is unique - even if two rows contain otherwise similar data. They are incredibly important, because without them, the only thing that's determining differences between your rows is simply the content within them. If you have more than one row with identical values that have NO primary key, there's no way to distinguish between them.
 
-Researched answer:
+Researched answer: In addition to my previous answer, it's a fun fact to note that there are actually two types of primary keys: natural keys, and surrogate keys. Natural keys are a column (or a set of columns) that already exist in the table and will be a unique value, whereas a surrogate key is in itself a unique value and is usually automatically generated.
```

Comment:
You did excellent in this section!  The only correction I would have had for you, you caught yourself.  Great work 🎉